### ISSUE TITLE
Fallback from webread()/webwrite() to urlread()/urlwrite()

### ISSUE THREAD

**Issue by benjamin-heasly** [isetbio/RemoteDataToolbox#16]
Currently toolbox functions use the recommended, built-in `webread()` and `webwrite()` functions for server queries.  This is fine for Matlab 2015b and later.

For matlab 2015a and earlier, `webread()` and `webwrite()` have slightly inconsistent behavior, or may not exist at all.  But we can still implement queries using the older `urlread()` and `urlwrite()` functions.

Modify `rdtRequestWeb()` with a version-dependent fallback to the `url*` functions for 2015a and earlier.  

Make the fallback unit-testable by adding an optional version parameter with the behavior, "try to run as though the Matlab were at the given version".

**Comment by benjamin-heasly** [comment]
It turns our `urlread()` and `urlwrite()` won't work for our purposes.

For some queries, we need to be able to POST a JSON request body, but `url*` only let us POST form fields as request headers.

We also would like to specify additional headers which `url*` don't allow us to configure:
- Content-Type for POST request body
- Accept type for GET and POST response body

See article about [`url*` limitations](http://undocumentedmatlab.com/blog/expanding-urlreads-capabilities).

The article mentions [`urlread2`](http://www.mathworks.com/matlabcentral/fileexchange/35693-urlread2), which may give us what we need.  Will try it and report back...

**Comment by benjamin-heasly** [comment]
`urlread2` eventually worked.  In `rdtRequestWeb` I had to accommodate two shortcomings of `urlread2`:
- Unlike `webread()`, `urlread2()` assumes query parameters are always strings.  I wrote a query encoder that automatically converts numeric and logical parameters.
- Unlike `webread()`, `urlread2()` did not help us with Basic Authentication.  I added request headers for Basic Authentication.
